From 5dd63703289a2870c381d4a5b839030bdc82211a Mon Sep 17 00:00:00 2001
From: Linus Torvalds <torvalds@linux-foundation.org>
Date: Fri, 3 Jun 2022 15:46:03 -0700
Subject: [PATCH 21/21] firmware_loader: enable XZ by default if compressed
 support is enabled

Commit 23cfbc6ec44e ("firmware: Add the support for ZSTD-compressed
firmware files") added support for ZSTD compression, but in the process
also made the previously default XZ compression a config option.

That means that anybody who upgrades their kernel and does a

    make oldconfig

to update their configuration, will end up without the XZ compression
that the configuration used to have.

Add the 'default y' to make sure this doesn't happen.

The whole compression question should probably be improved upon, since
it is now possible to "enable" compression in the kernel config but not
enable any actual compression algorithm, which makes it all very
useless.  It makes no sense to ask Kconfig questions that enable
situations that are nonsensical like that.

This at least fixes the immediate problem of a kernel update resulting
in a nonbootable machine because of a missed option.

Fixes: 23cfbc6ec44e ("firmware: Add the support for ZSTD-compressed  firmware files")
Cc: Takashi Iwai <tiwai@suse.de>
Cc: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>
---
 drivers/base/firmware_loader/Kconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/base/firmware_loader/Kconfig b/drivers/base/firmware_loader/Kconfig
index 7e663dd97..5166b323a 100644
--- a/drivers/base/firmware_loader/Kconfig
+++ b/drivers/base/firmware_loader/Kconfig
@@ -177,6 +177,7 @@ config FW_LOADER_COMPRESS_XZ
 	bool "Enable XZ-compressed firmware support"
 	select FW_LOADER_PAGED_BUF
 	select XZ_DEC
+	default y
 	help
 	  This option adds the support for XZ-compressed files.
 	  The files have to be compressed with either none or crc32
-- 
2.36.1.203.g1bcf4f6271

